### PR TITLE
Add ruby libraries and more to docker ci images

### PR DESCRIPTION
**Pull request by gaiksaya**
Signed-off-by: Sayali Gaikawad <gaiksaya@amazon.com>

### Description
Need ruby to build client libraries. Changes copied from @peterzhuamazon 's fork : https://github.com/peterzhuamazon/opensearch-build/tree/opensearch-rpm-packaging/docker/ci/dockerfiles
 
### Issues Resolved
Part of future releases
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: docker/ci/dockerfiles/ci-runner.al2.dockerfile [hunk 1]
```diff
@@ -86,16 +92,25 @@ RUN export MAVEN_URL=`curl -s https://maven.apache.org/download.cgi | grep -Eo '
...
+RUN curl -sSL https://rvm.io/mpapis.asc | gpg2 --import - && \
+    curl -sSL https://rvm.io/pkuczynski.asc | gpg2 --import - && \
+    curl -sSL https://get.rvm.io | bash -s stable
+
+# Switch shell for rvm related commands
+SHELL ["/bin/bash", "-lc"]
+CMD ["/bin/bash", "-l"]
+
+# Install ruby / rpm / fpm related dependencies
+RUN . /etc/profile.d/rvm.sh && rvm install 2.3.3 && rvm --default use 2.3.3 && \
```
Comment (peterzhuamazon) [inline docker/ci/dockerfiles/ci-runner.al2.dockerfile:110]: RVM does not have 2.3.3 precompiled binary like CentOS7.
https://rvm_io.global.ssl.fastly.net/binaries/amazon/2/x86_64/

Suggest adding this as 2.4.0 version.

Thanks.

**Comment by peterzhuamazon** [comment]
Bare in mind the ruby version here are bare minimum for fpm to install and we can increase the versions later on if reach LTS.

Thanks.

**Comment by gaiksaya** [comment]
> Bare in mind the ruby version here are bare minimum for fpm to install and we can increase the versions later on if reach LTS.
> 
> Thanks.

@VijayanB will this version be an issue for ruby clients?

File: docker/ci/dockerfiles/ci-runner.al2.dockerfile (hunk 1 above)
Comment (gaiksaya) [reply docker/ci/dockerfiles/ci-runner.al2.dockerfile]: Changed to 2.4.0. Thanks!

**Review by peterzhuamazon — APPROVED** [review]
Thanks.

**Comment by VijayanB** [comment]
@gaiksaya We will be using only gem. Hence, if we run following command before build, it will update to latest version
```
gem update --system
```

**Comment by abhinavGupta16** [comment]
The code coverage seems to have dropped. Can that be improved?

**Review by bbarani — APPROVED** [review]

**Comment by gaiksaya** [comment]
> The code coverage seems to have dropped. Can that be improved for dockerfile?

I dont see tests for docker image and this is CI image so not sure.

**Pull request merged**